infrastructure/Scripts/trafficmanager-update.ps1

```diff
@@ -0,0 +1,48 @@
+﻿# Parameters defined
+param(
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string]$Subscription,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string]$TrafficManager,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string]$TrafficManagerResourceGroup,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string]$Endpoint,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string]$EndpointType,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [bool]$TrafficEnabled
+)
+
+
+Write-Host "Setting subscription context: " $Subscription
+#set the subscripton context
+Set-AzContext -SubscriptionId $Subscription
+
+Write-Host "Get TrafficManager Details for: " $TrafficManager
+$trafficmanagerDetails = Get-AzTrafficManagerProfile -ResourceGroupName $TrafficManagerResourceGroup -Name $TrafficManager
+
+if($trafficmanagerDetails) {
+
+    if ($TrafficEnabled){
+        Enable-AzTrafficManagerEndpoint -Name $Endpoint -Type $EndpointType -ProfileName $TrafficManager -ResourceGroupName $TrafficManagerResourceGroup 
+    }
+    else {
+        Disable-AzTrafficManagerEndpoint -Name $Endpoint -Type $EndpointType -ProfileName $TrafficManager -ResourceGroupName $TrafficManagerResourceGroup -Force
+    }
+    Write-Host "Updated trafficmanager settings for " $TrafficManager
+
+} else {
+    Write-Host "Could not retrieve trafficmanager details for: " $TrafficManager
```

Comment:
How does this script exit with non 0 code i.e. indicating failure and hence stopping the devops flow?

Reply:
@ShekharGupta1988 Correct me if I am wrong.  Since there is no exception handling in this script, it will throw an error if an exception occurs (Say 401 from Azure) and return with non 0 code?

Reply:
good point. I think we should try it out once just running the script against a non-accessible TM (401 scenario) or non-existent TM (404 scenario) and see the error code.

Reply:
@DarrelDonald can you try it out against a failing resource and see if the script returns with non 0 exit code?